jerry-core/jerry-api.h

```diff
@@ -77,51 +89,110 @@ typedef uint32_t jerry_size_t;
+
+  /** [[Writable]] */
+  bool is_writable;
+
+  /** Is [[Enumerable]] defined? */
+  bool is_enumerable_defined;
+
+  /** [[Enumerable]] */
+  bool is_enumerable;
+
+  /** Is [[Configurable]] defined? */
+  bool is_configurable_defined;
+
+  /** [[Configurable]] */
+  bool is_configurable;
+
+  /** [[Value]] */
+  jerry_value_t value;
+
+  /** [[Get]] */
+  jerry_value_t getter;
+
+  /** [[Set]] */
+  jerry_value_t setter;
+} jerry_property_descriptor_t;
 
 /**
  * Type of an external function handler
  */
-typedef bool (*jerry_external_handler_t) (const jerry_object_t *function_obj_p,
-                                          const jerry_value_t this_val,
-                                          const jerry_value_t args_p[],
-                                          const jerry_length_t args_count,
-                                          jerry_value_t *ret_val_p);
+typedef jerry_value_t (*jerry_external_handler_t) (const jerry_value_t function_obj_p,
+                                                   const jerry_value_t this_val,
+                                                   const jerry_value_t args_p[],
+                                                   const jerry_length_t args_count);
 
 /**
  * Native free callback of an object
  */
 typedef void (*jerry_object_free_callback_t) (const uintptr_t native_p);
 
 /**
- * Function type applied for each fields in objects
+ * Function type applied for each data property of an object
+ */
+typedef bool (*jerry_object_property_foreach_t) (const jerry_value_t property_name_p,
+                                                 const jerry_value_t property_value,
+                                                 void *user_data_p);
+
+
+/**
+ * Logger functions
  */
-typedef bool (*jerry_object_field_foreach_t) (const jerry_string_t *field_name_p,
-                                              const jerry_value_t field_value,
-                                              void *user_data_p);
+#ifdef JERRY_ENABLE_LOG
+extern int jerry_debug_level;
+extern FILE *jerry_log_file;
+#endif /* JERRY_ENABLE_LOG */
+
+/**
+ * General engine functions
+ */
+void jerry_init (jerry_init_flag_t);
+void jerry_cleanup (void);
+void jerry_register_magic_strings (const jerry_char_ptr_t *, uint32_t, const jerry_length_t *);
+void jerry_get_memory_limits (size_t *, size_t *);
+void jerry_gc (void);
+
+/**
+ * Parser and executor functions
+ */
+bool jerry_run_simple (const jerry_char_t *, size_t, jerry_init_flag_t);
+jerry_value_t jerry_parse (const jerry_char_t *, size_t);
+jerry_value_t jerry_run (void);
+jerry_value_t jerry_eval (const jerry_char_t *, size_t, bool);
 
 /**
  * Get the global context
  */
-jerry_object_t *jerry_get_global (void);
+jerry_value_t jerry_get_global (void);
 
 /**
  * Checker functions of 'jerry_value_t'
  */
+bool jerry_value_is_array (const jerry_value_t);
 bool jerry_value_is_boolean (const jerry_value_t);
-bool jerry_value_is_error (const jerry_value_t);
```

Comment:
Yes, it is confusing that Error type and value with error flag.  
1) the new apis return jerry_value_t instead of a boolean, and we should check if it run correctly by jerry_value_has_error_flag(), right?
2) But the api: jerry_create_error, it will set the error flag after create the error object. How should we create a pure error type object